packages/react-data-grid/src/common/editors/__tests__/EditorContainer.spec.tsx

```diff
@@ -55,18 +64,8 @@ describe('Editor Container Tests', () => {
   describe('Basic render tests', () => {
     it('should select the text of the default input when the editor is rendered', () => {
       const { wrapper } = setup();
-      const isTextSelected = (input) => {
-        if (typeof input.selectionStart === 'number') {
-          return input.selectionStart === 0 && input.selectionEnd === input.value.length;
-        }
-        if (typeof document.selection !== 'undefined') {
-          input.focus();
-          return document.selection.createRange().text === input.value;
-        }
-      };
-      const editorNode = wrapper.instance().getInputNode();
-
-      expect(isTextSelected(editorNode)).toBeDefined();
+      const input = wrapper.instance().getInputNode() as HTMLInputElement;
```

Comment:
I think we can add the return type to `getInputNode` directly, create an interface and let the editor implements it.

e.g.
```
interface IEditorBase<T> {
  getInputNode: () => HTMLInputElement
}

class EditorContainer extends React.Component implements IEditorBase
```
Then we can save this `as` cast. `IEditorBase` can be further extended by `IEditor` and add `getValue` for example

Reply:
https://github.com/adazzle/react-data-grid/blob/next/packages/react-data-grid/src/common/types.ts#L85
We do have an interface for this already, but `getInputNode` might not necessarily return an input element.
I think it's fine to leave that kind of loose type-casting in tests, as it's not always easy to make TS happy, especially with the enzyme wrappers.

Reply:
@amanmahajan7 Could you confirm this? I remember it does always return the input (or maybe add select?) element ref so that we can set focus on that.

Reply:
agree that we should get rid of this method, it's confusing.